source/particle/world.cc

```diff
@@ -1606,18 +1644,32 @@ namespace aspect
       generator->parse_parameters(prm);
       generator->initialize();
 
-      // Create an output object depending on what the parameters specify
-      output.reset(Output::create_particle_output<dim>
-                   (prm));
-
-      // We allow to not generate any output plugin, in which case output is
-      // a null pointer. Only initialize output if it was created.
-      if (output)
+      // Remove duplicate output formats specified in the parameter file.
+      output_format_names.erase(std::unique(output_format_names.begin(), output_format_names.end()),
+                                output_format_names.end());
+
+      // Create an output object depending on what the parameters specify so long as none has not been specified.
+      // If none and more than one other format is specified, then we ignore none.
+      AssertThrow ((std::find (output_format_names.begin(),
+                               output_format_names.end(),
+                               "none") == output_format_names.end())
+                   ||
+                   (output_format_names.size() == 1),
+                   ExcMessage ("If you specify 'none' for the output format for parameter \"Data output format\","
+                               "then this needs to be the only value given."));
+
+      if (output_format_names.size() != 1 || output_format_names[0] != "none")
+        for (std::vector<std::string>::const_iterator itr = output_format_names.begin();
+             itr != output_format_names.end(); itr++)
```

Comment:
Again please rename the iterators in the two for-loops. `output_format` and `output_object` come to mind.